tests/unit/utils/test_integration_utils.py

```diff
@@ -0,0 +1,42 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from zenml.utils.integration_utils import parse_requirement
+
+
+def test_parse_requirement():
```

Comment:
Other suggestions that I'd maybe implement tests for (and maybe original function needs updates to handle certain errors around bad inputs):

### Edge Case
1. **Multiple extras**: "package[extra1,extra2]" should return `("package", "[extra1,extra2]")`

### Invalid Cases
1. **Empty string**: "" should return `(None, None)`
2. **Just extras without name**: "[extra]" should return `(None, None)`
3. **Only digits**: "123" should return `("123", None)`
4. **Non-alphanumeric characters in name**: "package!" should return `(None, None)`
5. **Non-alphanumeric characters in extras**: "package[extra!]" should return `(None, None)`
6. **Spaces in name**: "package name" should return `(None, None)`
7. **Spaces in extras**: "package[extra 1]" should return `(None, None)`

### Misc Cases
1. **Leading and trailing whitespaces**: " package " should probably return `(None, None)` unless you decide to `.strip()` the input.